typing/typeopt.ml

```diff
@@ -168,9 +168,41 @@ let value_kind env ty =
       Pboxedintval Pint64
   | Tconstr(p, _, _) when Path.same p Predef.path_nativeint ->
       Pboxedintval Pnativeint
+  | Tconstr(p, _, _) ->
+      if Numbers.Int.Set.mem ty.id visited then
+        Pgenval
+      else begin
+        match Env.find_type_descrs p env with
+        | exception Not_found -> Pgenval
+        | [], _ -> Pgenval
```

Comment:
This appears to  be handling the case of variant types with a single constructor, but not handling the more important case of record types.

Reply:
Thanks for noticing. I think this is fixed but I had to use `find_type` rather than `find_type_descr`. It seems more robust.